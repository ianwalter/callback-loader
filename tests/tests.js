const { join } = require('path')
const { test } = require('@ianwalter/bff')
const webpack = require('webpack')
const MemoryFileSystem = require('memory-fs')

test('loader executes callback with received content', ({ expect }) => {
  return new Promise(resolve => {
    const compiler = webpack({
      entry: join(__dirname, './fixtures/example.js'),
      module: {
        rules: [
          {
            test: /\.js$/,
            loader: join(__dirname, '../'),
            options: {
              callback: content => {
                expect(content).toMatchSnapshot()
                resolve()
              }
            }
          }
        ]
      }
    })
    compiler.outputFileSystem = new MemoryFileSystem()
    compiler.run()
  })
})
