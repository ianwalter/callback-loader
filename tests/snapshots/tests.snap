// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader executes callback with received content 1`] = `
"module.exports = { message: 'Hello!' }
"
`;
